src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java

```diff
@@ -23,8 +45,71 @@ public RewardScoreEntity recalculateScore(AllRewardScoresEntity allRewardScores,
     public RewardScoreEntity calculateOnContentWorkedOn(AllRewardScoresEntity allRewardScores, List<Content> contents, UserProgressLogEvent event) {
         RewardScoreEntity fitnessScoreBefore = allRewardScores.getFitness();
 
-        // TODO calculate fitness score
+        double fitnessDecrease = calculateFitnessDecrease(contents,event);
+        double newFitnessScore = fitnessScoreBefore.getValue() * (1 - fitnessDecrease);
+        int IntnewFitnessScore= (int) Math.round(newFitnessScore);
+        RewardScoreLogEntry logEntry = RewardScoreLogEntry.builder()
+                .date(OffsetDateTime.now())
+                .difference(fitnessScoreBefore.getValue() - IntnewFitnessScore)
+                .oldValue(fitnessScoreBefore.getValue())
+                .newValue(IntnewFitnessScore)
+                .reason(RewardChangeReason.CONTENT_DONE)
+                .associatedContentIds(Collections.emptyList())
+                .build();
+
+        fitnessScoreBefore.setValue(IntnewFitnessScore);
+        fitnessScoreBefore.getLog().add(logEntry);
 
         return fitnessScoreBefore;
     }
+
+    private double calculateFitnessDecrease(List<Content> contents, UserProgressLogEvent event) {
+        double fitnessDecrease = 0.0;
+
+        for (Content content : contents) {
+            if (isDueForRepetition(content)) {
+                int daysOverdue = calculateDaysOverdue(content);
+                double correctness = calculateCorrectness(event);
+                double decreasePerDay = 1 + (2 * daysOverdue * (1 - Math.pow(correctness, 2)));
+                fitnessDecrease += decreasePerDay;
+            }
+        }
+        return Math.min(MAX_DECREASE_PER_DAY, fitnessDecrease);
+    }
+
+    private boolean isDueForRepetition(Content content) {
+        // Implement the logic to determine if the content is due for repetition
+        // Return true if it is due, false otherwise
+        OffsetDateTime today = OffsetDateTime.now();
+        OffsetDateTime repetitionDate = content.getMetadata().getSuggestedDate();
+
+        // Check if the repetition date is today or in the past
+        boolean isDueForRepetition = repetitionDate.isBefore(today) || repetitionDate.isEqual(today);
+
+        return isDueForRepetition;
+    }
+    private int calculateDaysOverdue(Content content) {
+
+        OffsetDateTime today = OffsetDateTime.now();
+        OffsetDateTime repetitionDate = content.getMetadata().getSuggestedDate();
+
+        // Calculate the number of days between the current date and the repetition date
+        long daysBetween = ChronoUnit.DAYS.between(repetitionDate, today);
+
+        // If the content is due for repetition today, set daysOverdue to 1
+        int daysOverdue = daysBetween > 0 ? (int) daysBetween : 1;
+
+        return daysOverdue;
+    }
+    private double calculateCorrectness(UserProgressLogEvent event) {
```

Comment:
i would rename this slightly to `calculateCorrectnessModifier`

Reply:
done